### PR TITLE
dev_env_mac.md: Fix Java dependency

### LINKED ISSUE #944: adoptopenjdk vs. java in Development Environment on Mac guide
Following the installation guide for the dev tools on Mac, the guide gives the following instruction to resolve the Java (and Quartz) dependency:
`brew cask install xquartz java`

Brew, however, gave me the following instruction (for Java):
`brew cask install adoptopenjdk`

If adoptopenjdk is the right package, you might want to update the guide. Furthermore, moving the dependencies up would be helpful (for people like me, who don't read ahead ;) ).

https://dev.px4.io/master/en/setup/dev_env_mac.html

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #944

@julianoes FYI, I have changed this from "you may have an error so try install this" to "install all this for simulation to work". I think that makes sense because we have specific version requirements.

### CODE REVIEW

File: en/setup/dev_env_mac.md [hunk 1]
```diff
@@ -31,14 +31,22 @@ After installing Homebrew, run these commands in your shell to install the commo
...
+To support simulation you you should also install
+```sh
 brew install px4-sim
 ```
-
-If the installation outputs an error message about missing requirements follow the instructions. Your system will be missing Java and Quartz:
-
 ```sh
-brew cask install xquartz java
+# Quartz
```
Comment (hamishwillee) [inline en/setup/dev_env_mac.md:40]: @julianoes Do we still need quartz?

**Comment by julianoes** [comment]
I had some more changes in mind actually and I already had some local changes. Sorry, I forgot to push them earlier.

https://github.com/PX4/homebrew-px4/pull/50

File: en/setup/dev_env_mac.md (hunk 1 above)
Comment (julianoes) [reply en/setup/dev_env_mac.md:40]: That's a good question. Let me try without.

Comment (julianoes) [reply en/setup/dev_env_mac.md:40]: I can't try right now but let's just assume it's not needed anymore. PX4 doesn't need it and the Gazebo docs don't mention it either.

Comment (julianoes) [reply en/setup/dev_env_mac.md:40]: Hm, the docs here say it's still needed: https://raw.githubusercontent.com/Homebrew/install/master/install

**Comment by hamishwillee** [comment]
@julianoes Can you just make whatever changes need to now be made and merge?  Or abandon this and put in your own load changes? Sorry I jumped on this too early.

**Comment by julianoes** [comment]
@hamishwillee yes will do! Thx and sorry I didn't push it.

**Comment by hamishwillee** [comment]
Replaced

**Pull request closed without merging**